src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -234,6 +234,19 @@ internal BoundExpression BindRValueWithoutTargetType(ExpressionSyntax node, Diag
             return BindToNaturalType(BindValue(node, diagnostics, BindValueKind.RValue), diagnostics, reportDefaultMissingType);
         }
 
+        internal BoundExpression BindToTypeForErrorRecovery(BoundExpression expression, TypeSymbol type = null)
+        {
+            if (expression is null)
+                return null;
+            var discardedDiagnostics = DiagnosticBag.GetInstance();
+            var result =
+                (!expression.NeedsToBeConverted() || expression.WasConverted) ? expression :
+                type is null ? BindToNaturalType(expression, discardedDiagnostics) :
```

Comment:
>BindToNaturalType(expression, discardedDiagnostics) [](start = 31, length = 51)

```reportDefaultMissingType=false```?

Reply:
That would have no observable effect.

Reply:
> That would have no observable effect.

But it would avoid unnecessary  creation of diagnostics.

---
In reply to: [328380506](https://github.com/dotnet/roslyn/pull/38750#discussion_r328380506) [](ancestors = 328380506)